Do not call runtime-checks api from the task sdk anymore

prabhusneha opened the pull request:
Closes: https://github.com/apache/airflow/issues/47988

This PR removes the `/task_instances/{uuid}/runtime-checks` API and as its consumption.

Commit:
remove runtime-checks api from task sdk

amoghrajesh commented on airflow-core/src/airflow/api_fastapi/execution_api/routes/task_instances.py:
```diff
@@ -38,7 +38,6 @@
     TIRescheduleStatePayload,
     TIRetryStatePayload,
     TIRunContext,
-    TIRuntimeCheckPayload,
```
Can you remove the definition of this from datamodels too?

prabhusneha replied:
Removed from datamodels

amoghrajesh commented on task-sdk/src/airflow/sdk/execution_time/comms.py:
```diff
@@ -306,10 +305,6 @@ class SkipDownstreamTasks(TISkippedDownstreamTasksStatePayload):
     type: Literal["SkipDownstreamTasks"] = "SkipDownstreamTasks"
 
 
-class RuntimeCheckOnTask(TIRuntimeCheckPayload):
-    type: Literal["RuntimeCheckOnTask"] = "RuntimeCheckOnTask"
```
Lets also remove: https://github.com/apache/airflow/pull/48125/files#diff-c076ff9d88eb776c2f874ea4fbb66c9d090e080ed5c9b80c2c36dcb3a51360d3R224-R226

prabhusneha replied:
Removed it from where it could be removed, seems to be required at other places.

amoghrajesh reviewed (approved):
Just 2 minor comments, looks good. Could you verify one thing though, that the API call isn't sent from the task runner anymore?

Commit:
Remove TIRuntimeCheckPayload datamodel

Commit:
Remove OKResponse where not required

Commit:
Rebase

Commit:
Merge branch 'main' into remove_runtime_checks_api

amoghrajesh reviewed (approved):
Changes look good, lets wait for the CI

Commit:
Merge branch 'main' into remove_runtime_checks_api

Commit:
Merge branch 'main' into remove_runtime_checks_api

ashb committed:
Bring back the removed API endpoint, but mark it as removed in later versions

Although right now we don't have any API clients calling the previous version
since we aren't released yet and could have simply deleted the endpoint
outright, this is a good opportunity to use this as a practice run for API
versioning

kaxil reviewed (approved):
re: Versioning

Reviewed, it will take some getting used too -- but look sane.

ashb commented on task-sdk/dev/generate_task_sdk_models.py:
```diff
@@ -78,7 +78,7 @@ def generate_file():
     app = InProcessExecutionAPI()
 
-    latest_version = app.app.versions.version_values[-1]
+    latest_version = app.app.versions.version_values[0]
```
This was a previous bug -- we want the most recent version (`[0]`) not the oldest version (`[-1]`)
